### PR TITLE
Ner build fixups

### PR DESCRIPTION
**Pull request by jr1221**
Fixups for NER build.  Mainly switch to WSL as it is closer to Linux. See accompanying [Cerb PR](https://github.com/Northeastern-Electric-Racing/Cerberus/pull/213).

### CODE REVIEW

**Comment by jr1221** [comment]
I coded this all in notepad lol python is funny.

File: ner_environment/build_system/build_system.py [hunk 1]
```diff
@@ -55,7 +58,53 @@ def clang(args):
 # ==============================================================================
 
 def debug(args):
-    pass
+
+    command = []
+    #if args.docker:
+    #    print("Dockerized openocd unsupported for gdb")
```
Comment (dyldonahue) [inline ner_environment/build_system/build_system.py:64]: can we clean out unused code comments

File: ner_environment/build_system/build_system.py [hunk 2]
```diff
@@ -55,7 +58,53 @@ def clang(args):
...
+    else:
+        command = command + ["-f", "interface/cmsis-dap.cfg"]
+
+    build_directory = os.path.join("build", "*.elf")
+    elf_files = glob.glob(build_directory)
+    if not elf_files:
+        print("Error: No ELF file found in ./build/")
+        sys.exit(1)
+
+    elf_file = os.path.basename(os.path.normpath(elf_files[0]))  # Take the first ELF file found
```
Comment (dyldonahue) [inline ner_environment/build_system/build_system.py:82]: whats this basename command for. I trust it works ig but i thought u could just directly do elf_files[0]

File: ner_environment/build_system/build_system.py [hunk 3]
```diff
@@ -257,21 +325,7 @@ def run_command(command, stream_output=False, exit_on_fail=True):
...
-                break
-            if output:
-                print(output.strip())
-
-        # Capture any remaining output
-        stderr_output = process.stderr.read()
-        if stderr_output:
-            print(stderr_output, file=sys.stderr)
-
+        process = subprocess.Popen(command, text=True)
```
Comment (dyldonahue) [inline ner_environment/build_system/build_system.py:328]: did we find a way to get this streaming to work without that while loop? i thought we added that explicitly to get some form of sreamed output

File: ner_environment/build_system/build_system.py [hunk 4]
```diff
@@ -294,6 +348,18 @@ def disconnect_usbip():
     command = ["sudo", "usbip", "detach", "-p", "0"]
     run_command(command, exit_on_fail=False)
 
+def is_wsl(v: str = platform.uname().release) -> int:
+    """
+    detects if Python is running in WSL
```
Comment (dyldonahue) [inline ner_environment/build_system/build_system.py:353]: there are better ways to detect WSL: 

  if platform.system() == "Linux" and "microsoft" in platform.uname().release.lower():
  
  
  or 
  
   if 'WSL_DISTRO_NAME' in os.environ:
        return 2  # WSL2
        
        id rather us use one of these, seems cleaner than ur method

**Review by dyldonahue — CHANGES_REQUESTED** [review]

File: ner_environment/build_system/build_system.py (hunk 2 above)
Comment (jr1221) [reply ner_environment/build_system/build_system.py:82]: The filepath is different inside of docker.

File: ner_environment/build_system/build_system.py (hunk 3 above)
Comment (jr1221) [reply ner_environment/build_system/build_system.py:328]: Just don't use PIPE.  Everything apparently works as long as the code doesn't need to access the output.

Comment (dyldonahue) [reply ner_environment/build_system/build_system.py:328]: Idk I tested before adding this and running flash showed nothing on output, but if u tested this then alright

**Pull request merged**